src/Engine/TesseractEngine.php

```diff
@@ -0,0 +1,48 @@
+<?php
+declare(strict_types = 1);
+
+namespace App\Engine;
+
+use App\Exception\OcrException;
+use Symfony\Component\HttpClient\Exception\ClientException;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+use thiagoalessio\TesseractOCR\FriendlyErrors;
+use thiagoalessio\TesseractOCR\TesseractOCR;
+
+class TesseractEngine extends EngineBase
+{
+
+    /** @var HttpClientInterface */
+    private $httpClient;
+
+    public function __construct(HttpClientInterface $httpClient)
+    {
+        $this->httpClient = $httpClient;
+    }
+
+    public function getText(string $imageUrl, ?string $lang = null): string
+    {
+        // Check the URL and fetch the image data.
+        $this->checkImageUrl($imageUrl);
+        $imageResponse = $this->httpClient->request('GET', $imageUrl);
+        try {
+            $imageContent = $imageResponse->getContent();
+        } catch (ClientException $exception) {
+            throw new OcrException('image-retrieval-failed', [$exception->getMessage()]);
+        }
+
+        // Sanitize the language code.
+        $cleanLang = preg_replace('/[a-zA-Z]+/', '', $lang);
+
+        // Run OCR.
+        $ocr = new TesseractOCR();
+        // @TODO Remove this after this is resolved: https://github.com/thiagoalessio/tesseract-ocr-for-php/issues/210
+        FriendlyErrors::checkTesseractPresence($ocr->command->executable);
```

Comment:
This means it runs `type tesseract` twice before actually running it, which seems a bit unnecessary to do on every single run. I might add a patch upstream.